src/index.js

```diff
@@ -166,19 +172,32 @@ exports.createUploadLink = ({
       })
 
       options.body = form
-    } else options.body = payload
+    } else {
+      // If requested, set method to GET if there are no mutations.
+      const definitionIsMutation = d =>
+        d.kind === 'OperationDefinition' && d.operation === 'mutation'
+      if (
+        useGETForQueries &&
+        !operation.query.definitions.some(definitionIsMutation)
+      )
+        options.method = 'GET'
+
+      if (options.method === 'GET') {
+        const { newURI, parseError } = rewriteURIForGET(uri, body)
+        if (parseError) return fromError(parseError)
+        uri = newURI
+      } else options.body = payload
+    }
 
     return new Observable(observer => {
       // If no abort controller signal was provided in fetch options, and the
       // environment supports the AbortController interface, create and use a
       // default abort controller.
       let abortController
       if (!options.signal) {
-        const { controller } = createSignalIfSupported()
-        if (controller) {
-          abortController = controller
-          options.signal = abortController.signal
-        }
+        const { controller, signal } = createSignalIfSupported()
```

Comment:
It’s better to make such a change in a separate PR. Feel free to correct me, but I don't think this change solves anything as `abortController.signal` points to the same thing anyway. I had deliberately avoided the unnecessary abstraction in the Apollo API.